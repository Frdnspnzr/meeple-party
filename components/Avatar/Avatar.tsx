import classNames from "classnames";
import Image from "next/image";
import { CSSProperties } from "react";
import styles from "./avatar.module.css";

export default function Avatar({
  image,
  name,
  className,
  style,
}: {
  image?: string | null;
  name: string;
  className?: string;
  style?: CSSProperties;
}) {
  if (!!image) {
    return (
      <Image
        src={image}
        width="40"
        height="40"
        alt={!!name ? name : "User avatar"}
        className={classNames([styles.image, className])}
        style={style}
        unoptimized
        referrerPolicy="no-referrer"
      />
    );
  } else {
    const color = getColor(name);
    return (
      <div
        className={classNames([styles.dummy, className])}
        style={{
          ...style,
          backgroundColor: `rgb(${color[0]},${color[1]},${color[2]})`,
        }}
      >
        {name[0]}
      </div>
    );
  }
}

function getColor(name: string) {
  const hash = hashCode(name);
  const r = (hash & 0xff0000) >> 16;
  const g = (hash & 0x00ff00) >> 8;
  const b = hash & 0x0000ff;
  return [r, g, b];
}

function hashCode(str: string): number {
  var h: number = 0;
  for (var i = 0; i < str.length; i++) {
    h = (h + 6700417 * str.charCodeAt(i)) & 0xffffff;
  }
  return h;
}
