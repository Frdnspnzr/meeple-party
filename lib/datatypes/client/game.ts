export interface Game {
  id: number;
  name: string;
  thumbnail?: string;
  image?: string;
  year: number;
  playingTime: number;
  minPlayers: number;
  maxPlayers: number;
  weight: number;
  BGGRating?: number;
  BGGRank?: number;
}
