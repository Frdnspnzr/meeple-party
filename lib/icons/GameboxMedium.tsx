export default function IconGameBoxMedium() {
  return <i className="bi bi-grid-3x3-gap-fill"></i>;
}
