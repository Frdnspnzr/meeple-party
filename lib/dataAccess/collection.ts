import { GameCollectionResult } from "@/app/api/v2/user/[userId]/collection/route";
import axios from "axios";
import { UserId } from "../datatypes/client/userProfile";

export default function getCollection(
  userId: UserId
): Promise<GameCollectionResult> {
  return axios
    .get<GameCollectionResult>(`/api/v2/user/${userId}/collection`)
    .then((response) => response.data);
}
