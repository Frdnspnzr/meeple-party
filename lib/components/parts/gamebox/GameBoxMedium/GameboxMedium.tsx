"use client";

import Avatar from "@/components/Avatar/Avatar";
import { GameCollectionStatus } from "@/datatypes/collection";
import { UserProfile } from "@/datatypes/userProfile";
import { useTranslation } from "@/i18n/client";
import classNames from "classnames";
import Image from "next/image";
import Link from "next/link";
import StatusButton from "../../StatusButton/StatusButton";
import GameboxProps from "../interface";
import styles from "./gameboxmedium.module.css";

export default function GameboxMedium({
  game,
  friendCollections,
  myCollection,
  updateStatus,
}: Readonly<GameboxProps>) {
  const { t } = useTranslation("game");

  const update = (status: Partial<GameCollectionStatus>) => {
    if (updateStatus) {
      updateStatus(status);
    }
  };

  return (
    <div className={classNames(styles.container)}>
      <div className={styles.gamebox}>
        <Link href={`/app/game/${game.id}`}>
          <div className={styles.imageBox}>
            {!!game.thumbnail && (
              <Image
                src={game.thumbnail}
                className={`card-img-top ${styles.image}`}
                alt={game.name}
                width="200"
                height="150"
                unoptimized
              />
            )}
          </div>
        </Link>
        <div className={styles.title}>
          <h3 className="card-title">
            <Link href={`/app/game/${game.id}`}>{game.name}</Link>
          </h3>
        </div>
        <div className={styles.info}>
          <div className={styles.infoBox}>
            <div className={styles.metric}>
              {game.minPlayers === game.maxPlayers ? (
                game.minPlayers
              ) : (
                <>
                  {game.minPlayers}-{game.maxPlayers}
                </>
              )}
            </div>
            <div className={styles.label}>{t("Attributes.Players")}</div>
          </div>
          <div className={styles.infoBox}>
            <div className={styles.metric}>{game.playingTime}</div>
            <div className={styles.label}>
              {t("Filters.Traits.PlayingTime")}
            </div>
          </div>
          <div className={styles.infoBox}>
            <div className={styles.metric}>{round(game.weight)}</div>
            <div className={styles.label}>{t("Attributes.Weight")}</div>
          </div>
        </div>
        <div className={styles.collectionbuttons}>
          <StatusButton
            status={"own"}
            className={styles.statusButton}
            active={myCollection.own}
            toggle={() =>
              update({
                own: !myCollection.own,
                wantToPlay: myCollection.wantToPlay,
                wishlist: myCollection.wishlist,
              })
            }
          />
          <StatusButton
            status={"wanttoplay"}
            className={styles.statusButton}
            active={myCollection.wantToPlay}
            toggle={() =>
              update({
                own: myCollection.own,
                wantToPlay: !myCollection.wantToPlay,
                wishlist: myCollection.wishlist,
              })
            }
          />
          <StatusButton
            status={"wishlist"}
            className={styles.statusButton}
            active={myCollection.wishlist}
            toggle={() =>
              update({
                own: myCollection.own,
                wantToPlay: myCollection.wantToPlay,
                wishlist: !myCollection.wishlist,
              })
            }
          />
        </div>
      </div>
      <Link href={`/app/game/${game.id}`} className={styles.friends}>
        <div className={styles.collection}>
          <UserList users={friendCollections.own} />
        </div>
        <div className={styles.collection}>
          <UserList users={friendCollections.wantToPlay} />
        </div>
        <div className={styles.collection}>
          <UserList users={friendCollections.wishlist} />
        </div>
      </Link>
    </div>
  );
}

function round(x: number): number {
  return Math.round(x * 10) / 10;
}

function UserList({ users }: Readonly<{ users: UserProfile[] }>) {
  return (
    <>
      {users.map((u, i) => (
        <CollectionAvatar user={u} index={i} key={u.id} />
      ))}
    </>
  );
}

function CollectionAvatar({
  user,
  index,
}: Readonly<{
  user: UserProfile;
  index: number;
}>) {
  return (
    <Avatar
      name={user.name || ""}
      image={user.image}
      key={user.id}
      className={styles.person}
      style={{ zIndex: index * -1 }}
    />
  );
}
