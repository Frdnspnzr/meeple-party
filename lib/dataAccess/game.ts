import { GameGetResult } from "@/app/api/v2/game/[gameId]/route";
import { Game as ServerGame } from "@prisma/client";
import axios from "axios";
import { Game } from "../datatypes/client/game";

export function getGame(gameId: number): Promise<Game> {
  return axios
    .get<GameGetResult>(`/api/v2/game/${gameId}`)
    .then((response) => response.data.game)
    .then(convertGame);
}

export function convertGame(serverGame: ServerGame): Game {
  return {
    id: serverGame.id,
    maxPlayers: serverGame.maxPlayers,
    minPlayers: serverGame.minPlayers,
    name: serverGame.name,
    playingTime: serverGame.playingTime,
    weight: serverGame.weight,
    year: serverGame.year,
    BGGRank: serverGame.BGGRank || undefined,
    BGGRating: serverGame.BGGRating || undefined,
    image: serverGame.image || undefined,
    thumbnail: serverGame.thumbnail || undefined,
  };
}
