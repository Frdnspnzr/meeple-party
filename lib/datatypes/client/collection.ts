import { GameId } from "./game";
import { UserId } from "./userProfile";

export interface Collection {
  user: UserId;
  own: GameId[];
  wantToPlay: GameId[];
  wishlist: GameId[];
}