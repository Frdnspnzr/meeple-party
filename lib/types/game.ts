import * as client from "@prisma/client";

/**
 * General game information known to Meeple Party
 */
export type Game = Omit<client.Game, "updatedAt">;

/**
 * Game information as found on BoardGameGeek
 */
export interface BggGame extends Omit<Game, "wikidataId"> {
  description: string;
  designers: string[];
  artists: string[];
  age: number;
}

/**
 * Game information with additional data
 */
export type ExpandedGame = Game & {
  names: Omit<client.AlternateGameName, "gameId">[];
};

/**
 * Unique ID of a game. Equal to that game's ID on BoardGameGeek.
 */
export type GameId = number;

/**
 * Game data as written to the database.
 */
export type PrismaGame = client.Game;

/**
 * Game data as wirtten to the database, including alternate names.
 */
export type PrismaGameWithNames = PrismaGame & {
  alternateNames: client.AlternateGameName[];
};

/**
 * Converts a {@link PrismaGame} to {@link Game}, dropping all superfluous attributes.
 * @param game Game in database format to convert.
 */
export function prismaGameToGame(game: client.Game): Game {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    wikidataId: game.wikidataId,
    year: game.year,
  };
}

/**
 * Converts a {@link PrismaGameWithNames} to {@link ExpandedGame}, renaming attirbutes and dropping all superfluous attributes.
 * @param game Game in database format to convert.
 */
export function prismaGameToExpandedGame(
  game: PrismaGameWithNames
): ExpandedGame {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    wikidataId: game.wikidataId,
    year: game.year,
    names: game.alternateNames.map((n) => ({
      language: n.language,
      name: n.name,
    })),
  };
}

/**
 * Converts a {@link BggGame} to {@link ExpandedGame}, dropping all superfluous attributes.
 * @param game Game from BoardGameGeek to convert.
 */
export function bggGameToExpandedGame(game: BggGame): ExpandedGame {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    wikidataId: null,
    year: game.year,
    names: [],
  };
}

/**
 * Converts a {@link ExpandedGame} to {@link Game}, dropping all superfluous attributes
 * @param game Game to convert.
 */
export function expandedGameToGame(game: ExpandedGame): Game {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    wikidataId: game.wikidataId,
    year: game.year,
  };
}

/**
 * Converts a {@link Game} to {@link ExpandedGame}, initializing all needed attributes empty or undefined.
 * @param game Game to convert.
 */
export function gameToExpandedGame(game: Game): ExpandedGame {
  return { ...game, names: [] };
}

/**
 * Converts a {@link ExpandedGame} to {@link BggGame}, dropping all superfluous attributes and initializing all additional attributes empty or undefined.
 * @param game Game to convert
 */
export function expandedGameToBggGame(game: ExpandedGame): BggGame {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    year: game.year,
    age: 0,
    artists: [],
    description: "",
    designers: [],
  };
}
