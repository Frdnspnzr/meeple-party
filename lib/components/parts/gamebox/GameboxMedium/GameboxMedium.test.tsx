import Avatar from "@/components/Avatar/Avatar";
import { useModal } from "@/context/modalContext";
import { GameCollectionStatus, StatusByUser } from "@/datatypes/collection";
import { Game } from "@/datatypes/game";
import {
  generateArray,
  generateBoolean,
  generateUserProfile,
  render,
} from "@/utility/test";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import GameboxMedium from "./GameboxMedium";

jest.mock("@/i18n/client");
jest.mock("@/lib/hooks/useCollectionStatus");
jest.mock("@/components/Avatar/Avatar", () => ({
  __esModule: true,
  namedExport: jest.fn(),
  default: jest.fn(),
}));
jest.mock("@/context/modalContext");

const game: Game = {
  id: 123,
  name: "Test game",
  BGGRank: 23,
  BGGRating: 8.4,
  minPlayers: 1,
  maxPlayers: 5,
  playingTime: 120,
  weight: 3.8,
  year: 2018,
  image: "testimage",
  thumbnail: "testthumbnail",
};
const friendCollections: StatusByUser = {
  own: generateArray(generateUserProfile),
  wantToPlay: generateArray(generateUserProfile),
  wishlist: generateArray(generateUserProfile),
};
const myCollection: GameCollectionStatus = {
  own: true,
  wantToPlay: false,
  wishlist: true,
};

describe("GameboxMedium", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
    jest.mocked(Avatar).mockImplementation(() => <>Avatar</>);
  });
  beforeEach(() => {
    jest.mocked(useModal).mockImplementation(() => ({
      open: () => {},
      close: () => {},
    }));
  });
  it("Matches snapshot", async () => {
    const { container: rendered } = render(
      <GameboxMedium
        game={game}
        friendCollections={friendCollections}
        myCollection={myCollection}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it("Matches snapshot for precise weight", async () => {
    const { container: rendered } = render(
      <GameboxMedium
        game={{ ...game, weight: Math.PI }}
        friendCollections={friendCollections}
        myCollection={myCollection}
      />
    );
    expect(rendered).toMatchSnapshot("default game");
  });
  test.each(["own", "wanttoplay", "wishlist"])(
    "calls update callback when status button for %s is clicked",
    async (status) => {
      const user = userEvent.setup();
      let actualStatus = null;
      const callback = (status: Partial<GameCollectionStatus>) => {
        actualStatus = status;
      };
      const randomCollection = {
        own: generateBoolean(),
        wantToPlay: generateBoolean(),
        wishlist: generateBoolean(),
      };

      render(
        <GameboxMedium
          game={game}
          friendCollections={friendCollections}
          myCollection={randomCollection}
          updateStatus={callback}
        />
      );
      const button = screen.getByRole("button", {
        name:
          status === "own"
            ? "States.Own"
            : status === "wanttoplay"
              ? "States.WantToPlay"
              : "States.Wishlist",
      });
      await user.click(button);

      expect(actualStatus).not.toBeNull();

      if (status === "own") {
        expect(actualStatus!.own).toBe(!randomCollection.own);
      } else if (status === "wanttoplay") {
        expect(actualStatus!.wantToPlay).toBe(!randomCollection.wantToPlay);
      } else if (status === "wishlist") {
        expect(actualStatus!.wishlist).toBe(!randomCollection.wishlist);
      }
    }
  );
});
