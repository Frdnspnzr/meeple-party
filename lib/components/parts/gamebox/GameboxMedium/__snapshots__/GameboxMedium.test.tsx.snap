// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameboxMedium Matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="container"
  >
    <div
      class="gamebox"
    >
      <a
        href="/app/game/123"
      >
        <div
          class="imageBox"
        >
          <img
            alt="Test game"
            class="card-img-top image"
            data-nimg="1"
            decoding="async"
            height="150"
            loading="lazy"
            src="testthumbnail"
            style="color: transparent;"
            width="200"
          />
        </div>
      </a>
      <div
        class="title"
      >
        <h3
          class="card-title"
        >
          <a
            href="/app/game/123"
          >
            Test game
          </a>
        </h3>
      </div>
      <div
        class="info"
      >
        <div
          class="infoBox"
        >
          <div
            class="metric"
          >
            1
            -
            5
          </div>
          <div
            class="label"
          >
            Attributes.Players
          </div>
        </div>
        <div
          class="infoBox"
        >
          <div
            class="metric"
          >
            120
          </div>
          <div
            class="label"
          >
            Filters.Traits.PlayingTime
          </div>
        </div>
        <div
          class="infoBox"
        >
          <div
            class="metric"
          >
            3.8
          </div>
          <div
            class="label"
          >
            Attributes.Weight
          </div>
        </div>
      </div>
      <div
        class="collectionbuttons"
      >
        <button
          aria-label="States.Own"
          class="button statusButton"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-box-seam-fill"
          />
        </button>
        <button
          aria-label="States.WantToPlay"
          class="button statusButton"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-dice-3-fill"
          />
        </button>
        <button
          aria-label="States.Wishlist"
          class="button statusButton"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-gift-fill"
          />
        </button>
      </div>
    </div>
    <a
      class="friends"
      href="/app/game/123"
    >
      <div
        class="collection"
      >
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
      </div>
      <div
        class="collection"
      >
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
      </div>
      <div
        class="collection"
      >
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
      </div>
    </a>
  </div>
</div>
`;

exports[`GameboxMedium Matches snapshot for precise weight: default game 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="container"
  >
    <div
      class="gamebox"
    >
      <a
        href="/app/game/123"
      >
        <div
          class="imageBox"
        >
          <img
            alt="Test game"
            class="card-img-top image"
            data-nimg="1"
            decoding="async"
            height="150"
            loading="lazy"
            src="testthumbnail"
            style="color: transparent;"
            width="200"
          />
        </div>
      </a>
      <div
        class="title"
      >
        <h3
          class="card-title"
        >
          <a
            href="/app/game/123"
          >
            Test game
          </a>
        </h3>
      </div>
      <div
        class="info"
      >
        <div
          class="infoBox"
        >
          <div
            class="metric"
          >
            1
            -
            5
          </div>
          <div
            class="label"
          >
            Attributes.Players
          </div>
        </div>
        <div
          class="infoBox"
        >
          <div
            class="metric"
          >
            120
          </div>
          <div
            class="label"
          >
            Filters.Traits.PlayingTime
          </div>
        </div>
        <div
          class="infoBox"
        >
          <div
            class="metric"
          >
            3.1
          </div>
          <div
            class="label"
          >
            Attributes.Weight
          </div>
        </div>
      </div>
      <div
        class="collectionbuttons"
      >
        <button
          aria-label="States.Own"
          class="button statusButton"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-box-seam-fill"
          />
        </button>
        <button
          aria-label="States.WantToPlay"
          class="button statusButton"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-dice-3-fill"
          />
        </button>
        <button
          aria-label="States.Wishlist"
          class="button statusButton"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-gift-fill"
          />
        </button>
      </div>
    </div>
    <a
      class="friends"
      href="/app/game/123"
    >
      <div
        class="collection"
      >
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
      </div>
      <div
        class="collection"
      >
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
      </div>
      <div
        class="collection"
      >
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
        Avatar
      </div>
    </a>
  </div>
</div>
`;
